client_code/atomic/contexts.py

```diff
@@ -0,0 +1,109 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2021 anvilistas
+
+from functools import partial
+
+from .constants import ACTION, IGNORE, RENDER, SELECTOR
+from .rendering import active, call_queued, log, queued
+
+__version__ = "0.0.1"
+
+
+# ADDERS
```

Comment:
I'm not a fan of these sorts of comments - it becomes too easy to add stuff later in the wrong place.

I'd rather get rid altogether or, there's good reason to group these functions, create a class and put them both in there as methods.